File: java/src/test/java/TransactionTypeTest.java
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

@Nested
class TransactionTypeTest {

    @Test
    void verifyTransactionTypePayment() {
        assertThat(TransactionType.valueOf("PAYMENT")).isEqualTo(TransactionType.PAYMENT);
    }

    @Test
    void verifyTransactionTypeReversal() {
        assertThat(TransactionType.valueOf("REVERSAL")).isEqualTo(TransactionType.REVERSAL);
    }
}

File: kotlin/src/test/kotlin/TransactionTypeTest.kt
import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test

internal class TransactionTypeTest {

    @Test
    fun `parse "PAYMENT" should be ok`() {
        TransactionType.valueOf("PAYMENT") shouldBe TransactionType.PAYMENT
    }

    @Test
    fun `parse "REVERSAL" should be ok`() {
        TransactionType.valueOf("REVERSAL") shouldBe TransactionType.REVERSAL
    }
}